en/simulation/hitl.md

```diff
@@ -1,101 +1,133 @@
+By contrast, HITL runs normal PX4 firmware in "HITL mode", on normal hardware. The simulation data enters the system at a different point than for SITL. Core modules like commander and sensors have HIL modes at startup that bypass some of the normal functionality. 
 
-Currently Gazebo, jMAVSim, or X-Plane and QGroundControl are supported. Follow either setup step below.
+In summary, HITL runs PX4 on the actual hardware using standard firmware, but SITL actually executes more of the standard system code.
 
-### Using Gazebo
-* Update the environment variables: 
-``` 
-cd <Firmware_clone>
-make posix_sitl_default gazebo
-source Tools/setup_gazebo.bash $(pwd) $(pwd)/build/posix_sitl_default
-```
 
-* Open the vehicle model's sdf file (e.g. Tools/sitl_gazebo/models/iris/iris.sdf). 
+## Setting up HITL
 
-* Under the mavlink_interface plugin section, change the `serialEnabled` and `hil_mode` parameters to `true`. 
+### PX4 Configuration
 
-![](../../assets/sim/hil_params.png)
+1. Connect the autopilot directly to *QGroundControl* via USB.
+1. Enable HITL Mode
+   1. Open **Setup > Safety** section.
+   1. Enable HITL mode by selecting **Enabled** from the *HITL Enabled* list:
 
-* Replace the `serialDevice` parameter (`/dev/ttyACM0`) if necessary.
+      ![QGroundControl HITL configuration](../../assets/gcs/qgc_hitl_config.png)
+1. Select Airframe
+   1. Open **Setup > Airframes**
+   1. Select a [compatible airframe](#compatible_airframe) you want to test.
+      Generally you'll select *HILStar* for Fixed Wing/X-Plane simulator and a *HIL QuadCopter* option for copters (and jMAVSim or Gazebo).
 
-* **Make sure QGroundControl is not running**
+      ![Select Airframe](../../assets/gcs/qgc_hil_config.png)
+1. Setup UDP
+   1. Under the *General* tab of the settings menu, uncheck all *AutoConnect* boxes except for **UDP**. 
 
-* Connect the autopilot to the computer and allow it a few seconds to boot.
+      ![QGC Auto-connect settings for HITL](../../assets/gcs/qgc_hitl_autoconnect.png)
+1. (Gazebo only) Set the `SYS_COMPANION` parameter to `921600` (see [PX4 User Guide > Parameters](https://docs.px4.io/en/advanced_config/parameters.html#finding-a-parameter) for instructions on how to change parameters).
 
-* Run Gazebo in HITL mode 
-```
-gazebo /Tools/sitl_gazebo/worlds/iris.world
-```
-* Then run QGroundControl - it will auto-connect.
+1. (Optional) Configure Joystick and Failsafe.
 
-### Using jMAVSim \(Quadrotor only\)
+   Set the following [parameters](https://docs.px4.io/en/advanced_config/parameters.html#finding-a-parameter) in order to use a joystick instead of an RC remote control transmitter:
+   * [COM_RC_IN_MODE](../advanced/parameter_reference.md#COM_RC_IN_MODE) to "Joystick/No RC Checks". This allows joystick input and disables RC input checks.
+   * [NAV_DLL_ACT](../advanced/parameter_reference.md#NAV_DLL_ACT) to "Disabled". This ensures that no RC failsafe actions interfere when not running HITL with a radio control.
```

Comment:
@arwagoner @dagar  is NAV_DLL_ACT always needed if you use COM_RC_IN_MODE - shouldn't this automatically be set if COM_RC_IN_MODE is set to ""Joystick/No RC Checks"? I ask mostly because the QGC docs for Joystick do not mention this parameter.

Reply:
With HIL and QGC (xplane or flightgear) data link loss ends up being irrelevant because the mavlink connection has to be there for the simulation data.

Otherwise it can be set to whatever's appropriate for the usage.

Reply:
OK, I think I'll just leave this in as it will "cause no harm".